src/test/java/com/thomashan/collection/immutable/ImmutableListImplTest.java

```diff
@@ -0,0 +1,254 @@
+package com.thomashan.collection.immutable;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.ListIterator;
+
+import static com.thomashan.collection.immutable.ImmutableListImpl.of;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ImmutableListImplTest {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This class has too many methods, consider refactoring it.](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4509131)